sentry-types/src/protocol/v7.rs

```diff
@@ -1595,6 +1595,113 @@ impl fmt::Display for Span {
     }
 }
 
+/// The status of a Span.
+#[derive(Serialize, Deserialize, Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub enum SpanStatus {
+    /// The operation completed successfully.
+    #[serde(rename = "ok")]
+    Ok,
+    /// Deadline expired before operation could complete.
+    #[serde(rename = "deadline_exceeded")]
+    DeadlineExceeded,
+    /// 401 Unauthorized (actually does mean unauthenticated according to RFC 7235)
+    #[serde(rename = "unauthenticated")]
+    Unauthenticated,
+    /// 403 Forbidden
+    #[serde(rename = "permission_denied")]
+    PermissionDenied,
+    /// 404 Not Found. Some requested entity (file or directory) was not found.
+    #[serde(rename = "not_found")]
+    NotFound,
+    /// 429 Too Many Requests
+    #[serde(rename = "resource_exhausted")]
+    ResourceExhausted,
+    /// Client specified an invalid argument. 4xx.
+    #[serde(rename = "invalid_argument")]
+    InvalidArgument,
+    /// 501 Not Implemented
+    #[serde(rename = "unimplemented")]
+    Unimplemented,
+    /// 503 Service Unavailable
+    #[serde(rename = "unavailable")]
+    Unavailable,
+    /// Other/generic 5xx.
+    #[serde(rename = "internal_error")]
+    InternalError,
+    /// Unknown. Any non-standard HTTP status code.
+    #[serde(rename = "unknown_error")]
+    UnknownError,
+    /// The operation was cancelled (typically by the user).
+    #[serde(rename = "cancelled")]
+    Cancelled,
+    /// Already exists (409)
+    #[serde(rename = "already_exists")]
+    AlreadyExists,
+    /// Operation was rejected because the system is not in a state required for the operation's
+    #[serde(rename = "failed_precondition")]
+    FailedPrecondition,
+    /// The operation was aborted, typically due to a concurrency issue.
+    #[serde(rename = "aborted")]
+    Aborted,
+    /// Operation was attempted past the valid range.
+    #[serde(rename = "out_of_range")]
+    OutOfRange,
+    /// Unrecoverable data loss or corruption
+    #[serde(rename = "data_loss")]
+    DataLoss,
+}
+
+impl str::FromStr for SpanStatus {
+    type Err = ParseLevelError;
```

Comment:
Maybe a more descriptive error would be a good idea. I have to look at more examples, I’m not sure what kind of errors we use for all the other enums here.

Reply:
I am sorry, going to add `ParseStatusError` struct and use it here.